http/interceptor.go

```diff
@@ -108,7 +108,22 @@ func (i *rwInterceptor) Header() http.Header {
 	return i.w.Header()
 }
 
-var _ http.ResponseWriter = (*rwInterceptor)(nil)
+func (i *rwInterceptor) ReadFrom(r io.Reader) (n int64, err error) {
+	return io.Copy(i, r)
+}
+
+func (i *rwInterceptor) Flush() {
+	// coraza middleware always needs to buffer the entire request, response cycle
```

Comment:
I wonder if we should drop a log here same way we do in https://github.com/corazawaf/coraza/pull/923/files#diff-110e39705812e7ff833d85e01bd91376daaf071b1bd9c458fc5ed9a0287a88a6L34

Reply:
This comment made me take a closer look at `http.Flusher` and there is one thing that can and should be in this implementation.

`WriteHeader` must be called if it wasn't called before with a status code of 200.

Thank you for pointing out that the was room for improvement here.